### ISSUE TITLE
Make client connection pooling policy more configurable

### ISSUE THREAD

**Issue by trustin** [line/armeria#1630]
A user wants to customize the client-side connection pooling policy, e.g.

- the number of minimum/maximum idle connections to keep, e.g. keep at least 3 idle connections.
- the number of connections to keep, e.g. have no more than N connections.
- ...

**Comment by kapsterio** [comment]
is someone working on this feature?

**Comment by hyangtack** [comment]
@kapsterio No, as I know. Are you interested in sending a PR for this feature?
